test/e2e/reconciler/partition/partitionset_test.go

```diff
@@ -0,0 +1,332 @@
+
+	t.Logf("Creating a shard matching the partitionSet")
+	shard1a := &corev1alpha1.Shard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "partition-shard-1a",
+			Labels: map[string]string{
+				"region": "partition-test-region-1",
+			},
+		},
+		Spec: corev1alpha1.ShardSpec{
+			BaseURL: "https://base.kcp.test.dev",
+		},
+	}
+	shardClient := kcpClusterClient.CoreV1alpha1().Shards()
+	_, err = shardClient.Cluster(core.RootCluster.Path()).Create(ctx, shard1a, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating shard")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) && partitionSet.Status.Count == uint16(1) {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", partitionSet.Status.Count)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected the partition count to be 1")
+	partitionClient := kcpClusterClient.TopologyV1alpha1().Partitions()
+	var partitions *topologyv1alpha1.PartitionList
+	framework.Eventually(t, func() (bool, string) {
+		partitions, err = partitionClient.Cluster(partitionClusterPath).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error retrieving partitions")
+		if len(partitions.Items) == 1 {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 1 partition, but got %d", len(partitions.Items))
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected 1 partition")
+
+	t.Logf("Creating a shard part of a second region")
+	shard2 := &corev1alpha1.Shard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "partition-shard-2",
+			Labels: map[string]string{
+				"region": "partition-test-region-2",
+			},
+		},
+		Spec: corev1alpha1.ShardSpec{
+			BaseURL: "https://base.kcp.test.dev",
+		},
+	}
+	shard2, err = shardClient.Cluster(core.RootCluster.Path()).Create(ctx, shard2, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating shard")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) && partitionSet.Status.Count == uint16(2) {
+			return true, ""
+		}
+		return false, fmt.Sprintf("expected 2 partitions, but got %d", partitionSet.Status.Count)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected the partitions to be ready and their count to be 2")
+	framework.Eventually(t, func() (bool, string) {
+		partitions, err = partitionClient.Cluster(partitionClusterPath).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error retrieving partitions")
+		if len(partitions.Items) == 2 {
+			return true, ""
```

Comment:
should we check `Partition.Spec.Selector` ?

Reply:
We need to validate that we get the right ones ehre